server/server.go

```diff
@@ -50,7 +50,11 @@ func (p *PTMergeServer) Run() {
 	}
 	log.Printf("Connected to mongodb at %s\n", p.DatabaseHost)
 
-	p.Database = session.DB(p.DatabaseName)
+	// this master database session should be copied using session.Copy()
+	// before making requests to the database. This protects the connection
+	// to mongo if for any reason a database operation times out.
```

Comment:
You say the `session` should be copied, but... I don't see it copied in `RegisterRoutes` or `MergeController`, but maybe that's because they don't actually _use_ it (they only assign it)?